invenio_records_resources/services/records/components.py

```diff
@@ -72,3 +51,40 @@ def create(self, identity, data=None, record=None, **kwargs):
     def update(self, identity, data=None, record=None, **kwargs):
         """Inject parsed metadata to the record."""
         record.metadata = data.get('metadata', {})
+
+
+class FilesOptionsComponent(ServiceComponent):
```

Comment:
The component should also deal with setting ``record.files.preview`` and ``record.files.order``.